### PR TITLE
Use SDK runtime boundary in apps

### PR DESCRIPTION
**Pull request by phroi**
## Why

Interface and tester still duplicated account projection, transaction completion, and confirmation polling that PR 29 moved behind the SDK boundary.

## Changes

- Move interface account projection, base transaction assembly, completion, and send confirmation onto SDK APIs
- Refresh interface transaction state immediately before wallet confirmation
- Extract tester runtime state and transaction building into an importable SDK-backed module
- Remove stale app-local migration surfaces that the SDK now owns
- Add app-level tests for the new SDK-owned runtime boundaries

### CODE REVIEW

File: apps/interface/src/Action.tsx [hunk 1]
```diff
@@ -122,50 +126,45 @@ export default function Action({
...
     setMessage("Transaction confirmed.");
     formReset();
     await new Promise((resolve) => setTimeout(resolve, 2000));
   } catch (error) {
     setFailure(errorMessageOf(error));
   } finally {
-    await queryClient.invalidateQueries({ queryKey: [chain, address] });
     freezeTxInfo(txInfoPadding);
+    await queryClient.invalidateQueries({ queryKey: l1StateQueryKey(walletConfig) });
+    await queryClient.invalidateQueries({ queryKey: [chain, address] });
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/Action.tsx]: Fixed in `b9d7eeb3fee29d90252f32ce90f4682c3d507f0e`. The broader `[chain, address]` invalidation already covers `l1StateQueryKey(walletConfig)`, so I removed the duplicate specific invalidation.

Validated with `pnpm --filter ./apps/interface test:ci`, `pnpm --filter ./apps/interface lint`, `pnpm --filter ./apps/interface build`, `git diff --check`, and `pnpm check`.

File: apps/interface/src/Action.tsx [hunk 2]
```diff
@@ -122,50 +126,44 @@ export default function Action({
...
-    ) {
-      setMessage("Waiting for network confirmation...");
-      await new Promise((resolve) => setTimeout(resolve, 10000));
-      status = (await cccClient.getTransaction(txHash))?.status;
-      checks += 1;
+    setMessage("Refreshing transaction...");
+    const txInfo = await buildFreshTxInfo();
+    if (txInfo.error !== "" || !hasTransactionActivity(txInfo.tx) || txInfo.fee <= 0n) {
+      throw new Error(txInfo.error || "Nothing to do right now");
     }
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/Action.tsx]: Fixed in `9525bc03bdb9ce58edece4e182f1b443185f09fb`. I split the rebuilt transaction checks so explicit transaction errors, empty transactions, and missing/invalid fees produce distinct messages.

Validated with `pnpm --filter ./apps/interface test:ci`, `pnpm --filter ./apps/interface lint`, `pnpm --filter ./apps/interface build`, `git diff --check`, and `pnpm check`.

File: apps/interface/src/transaction.ts [hunk 3]
```diff
@@ -175,14 +179,12 @@ async function buildIckbToCkbPreview(
     Math.min(candidates.length, MAX_WITHDRAWAL_REQUESTS),
     async (withdrawalCount) => {
       try {
-        // DAO withdrawal requests must claim matching input/output indexes, so
-        // build those pairs first and append the input-only base activity later.
-        let tx = ccc.Transaction.default();
+        let tx = baseTx.clone();
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/transaction.ts:182]: No code change here. I checked the CCC implementation used by this workspace: `ccc.Transaction` defines `clone()` as the deep-copy operation, while `ccc.Transaction.from(baseTx)` returns `baseTx` unchanged when `baseTx` is already a `Transaction`. Replacing `clone()` with `Transaction.from(baseTx)` would make preview attempts share and mutate the same base transaction.

Validated the current code with `pnpm --filter ./apps/interface test:ci`, `pnpm --filter ./apps/interface lint`, `pnpm --filter ./apps/interface build`, `pnpm --filter ./packages/sdk test:ci`, `git diff --check`, and `pnpm check`.

File: apps/interface/src/queries.ts [hunk 4]
```diff
@@ -58,71 +57,24 @@ export async function getL1State(
...
-    sumCkb(availableOrders);
-  const ickbAvailable =
-    ickbNative +
-    sumUdt(receipts) +
-    sumUdt(readyWithdrawals) +
-    sumUdt(availableOrders);
-
-  const ckbBalance = ckbAvailable + sumCkb(pendingWithdrawals) + sumCkb(pendingOrders);
-  const ickbBalance = ickbAvailable + sumUdt(pendingWithdrawals) + sumUdt(pendingOrders);
+  const projection = projectAccountAvailability(account, user.orders);
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/queries.ts]: Fixed in `24fb8b59c2356e3f582939a728c4c0e1c2ac0a85`. The interface now passes `{ collectedOrdersAvailable: true }` to preserve the previous balance semantics, and the `getL1State` regression test now asserts collected/matchable orders are projected as available.

Validated with `pnpm --filter ./apps/interface test:ci`, `pnpm --filter ./apps/interface lint`, `pnpm --filter ./apps/interface build`, `pnpm --filter ./packages/sdk test:ci`, `git diff --check`, and `pnpm check`.

File: apps/interface/src/transaction.ts [hunk 5]
```diff
@@ -144,7 +143,12 @@ async function buildCkbToIckbPreview(
         );
       }
 
-      return await finalizeTransaction(tx, estimatedMaturity, walletConfig);
+      return await finalizeTransaction(
+        tx,
+        estimatedMaturity,
+        context.system.feeRate,
+        walletConfig,
+      );
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/transaction.ts:151]: No code change here. This loop/fallback shape is not new in this PR: `origin/master` already finalized each candidate attempt inside `findBestAttempt`. Completion failures are the signal that the current candidate is not buildable and the preview should try a smaller direct-deposit count. Also, `tx.getFee` is reached only after `completeTransaction` succeeds, and `findBestAttempt` returns immediately on that first success.

File: apps/interface/src/transaction.ts [hunk 6]
```diff
@@ -231,7 +226,12 @@ async function buildIckbToCkbPreview(
           );
         }
 
-        return await finalizeTransaction(tx, estimatedMaturity, walletConfig);
+        return await finalizeTransaction(
+          tx,
+          estimatedMaturity,
+          context.system.feeRate,
+          walletConfig,
+        );
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/transaction.ts:234]: No code change here for the same reason as the CKB-to-iCKB path. The fallback loop must finalize each candidate because completion can fail for a specific withdrawal count, and that failure is what lets the preview try the next smaller candidate. This behavior predates the SDK migration; the PR moved the completion boundary into `sdk.completeTransaction` rather than adding a new loop.

File: apps/interface/src/Action.tsx [hunk 7]
```diff
@@ -81,12 +82,15 @@ export default function Action({
           onClick={() => {
             if (isStateStale) {
               void walletConfig.queryClient.invalidateQueries({
-                queryKey: [walletConfig.chain, walletConfig.address, "l1State"],
+                queryKey: l1StateQueryKey(walletConfig),
               });
               return;
             }
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/Action.tsx]: Fixed in `0a8a9527a07ca6c2311e2f5237599c014ac535e9`. I removed the stale-preview click short circuit and its prop plumbing because `transact` already rebuilds from fresh `getL1State(walletConfig)` before wallet confirmation.

File: apps/interface/src/Action.tsx [hunk 8]
```diff
@@ -122,50 +126,50 @@ export default function Action({
...
+
     setMessage("Transaction confirmed.");
     formReset();
     await new Promise((resolve) => setTimeout(resolve, 2000));
   } catch (error) {
     setFailure(errorMessageOf(error));
   } finally {
-    await queryClient.invalidateQueries({ queryKey: [chain, address] });
     freezeTxInfo(txInfoPadding);
+    await queryClient.invalidateQueries({ queryKey: [chain, address] });
```
[the remark this answers is not in the document]
Comment (phroi) [reply apps/interface/src/Action.tsx:161]: No code change here. Cleanup is already in `finally`, so the UI is unfrozen and the account query is invalidated on both success and failure. I kept the current order because `l1StateOptions(..., isFrozen)` disables the L1 query while frozen; unfreezing before invalidation lets the invalidated account-level query refetch normally.

**Comment by phroi** [comment]
LGTM

Phroi %179
